### PR TITLE
fix(form-field): Updates legend to be direct child of FieldSet

### LINKED ISSUE #1060: Grouped inputs do not announce legend to screen reader users
## 🐛 Bug Report

When a screen reader users moves focus to a grouped set of inputs the fieldset/legend relationship is not announced to a screen reader users. For example, if a group of checkboxes has a legend of "have you seen the new lego movie?" this will not be announced when focusing a checkbox with a label of "yes".

## To Reproduce

1. Navigate to story book
2. Open a grouped checkbox/radio button example
3. Turn on a screen reader (VO will work)
4. Navigate with arrow keys into the fieldset
5. Listen for no legend to be announced when entering the group or focusing a specific grouped input

## Expected Behavior

When moving focus into a fieldset using a screen reader users should hear both the legend text and label of the specific grouped input. 

## Actual Results

Currently the legend is not announced and there is no group announcement indication.

## Browser (if applicable)

Chrome

## Link to repl or repo (highly encouraged)

v4.7.0
radio group screen shot

### Error Output

Paste the error output here

```bash

```

or

Include a screenshot of a visual bug

### PR DESCRIPTION
**Pull request by sahlhoff**
## Summary

Fixes #1060 

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes
- [x] all (dev)dependencies that the module needs is added to its `package.json`
- [x] code has been documented and, if applicable, usage described in README.md
- [x] module has been added to `canvas-kit-react` and/or `canvas-kit-css` universal modules, if
      applicable
- [x] design approved final implementation
- [x] a11y approved final implementation
- [x] code adheres to the [API & Pattern guidelines](../modules/docs/mdx/API_PATTERN_GUIDELINES.mdx)

### CODE REVIEW

**Review by NicholasBoll — APPROVED** [review]

File: modules/react/form-field/lib/Label.tsx [hunk 1]
```diff
@@ -75,9 +75,18 @@ const RequiredAstrisk = styled('abbr')({
...
-}));
+const LegendComponent = styled('legend')<LabelProps>(...labelStyles, ({labelPosition}) => {
+  if (labelPosition === Label.Position.Top) {
+    return {
+      width: '100%',
+    };
+  }
+
+  return {
+    float: 'left',
```
Comment (NicholasBoll) [inline modules/react/form-field/lib/Label.tsx:86]: My only concern with `float` is use of a float will decrease rendering performance: https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing

I don't think this `float` will realistically tank rendering performance of a page and I'm sure many techniques were evaluated to get the correct result in all our supported browsers, but I wanted to make sure `flex` didn't work here first.

Comment (alanbsmith) [reply modules/react/form-field/lib/Label.tsx:86]: +1 I prefer flex over float if possible.

Comment (alanbsmith) [reply modules/react/form-field/lib/Label.tsx:86]: Okay, after diving into this, I realize that a `<legend>` element inside a `<fieldset>` doesn't support `display: flex`. Fun. I'd like to explore some other solutions, but not in this PR.

**Review by alanbsmith — APPROVED** [review]
Thanks, @sahlhoff!

File: modules/react/form-field/lib/Label.tsx (hunk 1 above)
Comment (sahlhoff) [reply modules/react/form-field/lib/Label.tsx:86]: I too prefer flex, but was unable to achieve the `Label.Position.Left` layout without it. @alanbsmith thankfully this is now  "supported" in modern browsers: https://bugzilla.mozilla.org/show_bug.cgi?id=1230207, https://bugs.chromium.org/p/chromium/issues/detail?id=375693 .... but still some weird edge cases that required the float – "fun" accurately describes this 😂

**Pull request merged**